polyfill/lib/absolute.mjs

```diff
@@ -1,126 +1,150 @@
-  let hour = ES.ISODateTimePartString(dateTime.hour);
-  let minute = ES.ISODateTimePartString(dateTime.minute);
-  let seconds = ES.ISOSecondsString(dateTime.second, dateTime.millisecond, dateTime.microsecond, dateTime.nanosecond);
-  let timeZoneString = ES.ISOTimeZoneString(timeZone, this);
-  let resultString = `${year}-${month}-${day}T${hour}:${minute}${seconds ? `:${seconds}` : ''}${timeZoneString}`;
-  return resultString;
-};
-Absolute.prototype.toLocaleString = function toLocaleString(...args) {
-  return new Intl.DateTimeFormat(...args).format(this);
-};
-Absolute.prototype.inZone = function inZone(timeZoneParam = 'UTC') {
-  const timeZone = ES.ToTimeZone(timeZoneParam);
-  return timeZone.getDateTimeFor(this);
-};
+    const result = GetSlot(this, EPOCHNANOSECONDS) - delta;
+    return new Absolute(result);
+  }
+  difference(other) {
+    other = ES.GetIntrinsic('%Temporal.absolute%')(other);
 
-Absolute.fromEpochSeconds = function fromEpochSeconds(epochSecondsParam) {
-  let epochSeconds = ES.ToNumber(epochSecondsParam);
-  let epochSecondsBigInt = BigInt(epochSeconds);
-  let epochNanoSeconds = epochSecondsBigInt * 1000000000n;
-  let resultObject = new Absolute(epochNanoSeconds);
-  return resultObject;
-};
-Absolute.fromEpochMilliseconds = function fromEpochMilliseconds(epochMillisecondsParam) {
-  let epochMilliseconds = ES.ToNumber(epochMillisecondsParam);
-  let epochMillisecondsBigInt = BigInt(epochMilliseconds);
-  let epochNanoSeconds = epochMillisecondsBigInt * 1000000n;
-  let resultObject = new Absolute(epochNanoSeconds);
-  return resultObject;
-};
-Absolute.fromEpochMicroseconds = function fromEpochMicroseconds(epochMicrosecondsParam) {
-  let epochMicroseconds = BigInt(epochMicrosecondsParam);
-  let epochNanoSeconds = epochMicroseconds * 1000n;
-  let resultObject = new Absolute(epochNanoSeconds);
-  return resultObject;
-};
-Absolute.fromEpochNanoseconds = function fromEpochNanoseconds(epochNanosecondsParam) {
-  let epochNanoseconds = BigInt(epochNanosecondsParam);
-  let resultObject = new Absolute(epochNanoseconds);
-  return resultObject;
-};
-Absolute.fromString = function fromString(isoString) {};
+    const [one, two] = [this, other].sort(Absoulte.compare);
+    const delta = two.getEpochNanoseconds() - one.getEpochNanoseconds();
+    const duration = ES.GetIntrinsic('%Temporal.duration%')(delta);
+    return duration;
+  }
+  toString(timeZoneParam = 'UTC') {
+    let timeZone = ES.GetIntrinsic('%Temporal.timezone%')(timeZoneParam);
+    let dateTime = timeZone.getDateTimeFor(this);
+    let year = ES.ISOYearString(dateTime.year);
+    let month = ES.ISODateTimePartString(dateTime.month);
+    let day = ES.ISODateTimePartString(dateTime.day);
+    let hour = ES.ISODateTimePartString(dateTime.hour);
+    let minute = ES.ISODateTimePartString(dateTime.minute);
+    let seconds = ES.ISOSecondsString(dateTime.second, dateTime.millisecond, dateTime.microsecond, dateTime.nanosecond);
```

Comment:
should all these be consts?

Reply:
This is a relict from trying to make the code read more like spec text

Reply:
that's fine, but then there's some other `const`s to convert to `let`s :-)